### ISSUE TITLE
Lists need cleanup parse-wise

### ISSUE THREAD

**Issue by kuerious** [chadmayfield/pihole-blocklists#13]
I was trying to troubleshoot an issue with my **pihole-FTL** daemon crashing about 1x/day. I ran across [this article](https://discourse.pi-hole.net/t/pi-hole-crashes/13920) and took a look inside of my blocklists. [Lo and behold](https://i.imgur.com/5KwyxWH.png), there it was:

https://i.imgur.com/5KwyxWH.png

I'm going to keep, but disable, your "all porn" blocklist for now. I'd edit the list myself, but a subsequent update would put them back.

Yes, I realize you made the list(s) as a PoC. I just thought I'd bring this to your attention.

**Comment by chadmayfield** [comment]
Thanks for writing up the issue!  That is not good that it may be causing your Pi-Hole to crash.

I'll take a look at the issue and see if it is truly that domain that is causing the issue.  It doesn't look like it is valid anymore or perhaps it's part of a CDN and the upstream provider parsed it incorrectly and now it begins with three dashes.

In the linked article it looks like there is also a "${tomcat_server}" and  "&quot;" before this domain.  I believe that is the actual issue since the domain you referenced is a domain has valid characters, the others are not and dnsmasq doesn't like them.   Regardless let me take a look at it and if needed I'll fix it or at the very least block that domain since it is no longer active.

**Comment by kuerious** [comment]
Happy to help. Doing so is the "open" part of open-source.

I counted the first 146 lines on your list as STARTING with a "-" symbol,
and the crashes are happening between 12:00M and 1:00am, which is I'm
guessing when the daily "gravity" checkup occurs.

The article I linked wasn't pointing to the " marks or Tomcat server, but
the leading "-" itself as a symbol which a domain cannot start with (but
can contain).
